src/Manticoresearch/Transport/UnixSocket.php

```diff
@@ -0,0 +1,217 @@
+use Manticoresearch\Exceptions\ResponseException;
+use Manticoresearch\Request;
+use Manticoresearch\Response;
+
+/**
+ * Class UnixSocket
+ * @package Manticoresearch\Transport
+ */
+class UnixSocket extends \Manticoresearch\Transport implements TransportInterface
+{
+    /**
+     * @var string
+     */
+    protected $scheme = 'http';
+    /**
+     * @var \Swoole\Client|null
+     */
+    protected static $client = null;
+
+    /**
+     * @param Request $request
+     * @param array   $params
+     *
+     * @return Response
+     */
+    public function execute(Request $request, $params = [])
+    {
+        $connection = $this->getConnection();
+        $client     = $this->getSocketConnection($connection->getConfig('persistent'));
+        $endpoint   = $request->getPath();
+
+        $url = $this->setupURI($endpoint, $request->getQuery());
+
+        $data      = $request->getBody();
+        $method    = $request->getMethod();
+        $headers   = $connection->getHeaders();
+        $headers[] = sprintf('Content-Type: %s', $request->getContentType());
+
+        if (!empty($data)) {
+            if (is_array($data)) {
+                $content = json_encode($data, JSON_THROW_ON_ERROR | JSON_UNESCAPED_UNICODE | JSON_UNESCAPED_SLASHES);
+            } else {
+                $content = $data;
+            }
+        }
+
+        if ($connection->getConnectTimeout() > 0) {
+            //TODO time out
+        }
+
+        if ($connection->getConfig('username') !== null && $connection->getConfig('password') !== null) {
+            //TODO: http auth
+            // $connection->getConfig('username')
+            // $connection->getConfig('password')
+        }
+
+        if ($connection->getConfig('proxy') !== null) {
+            //TODO http proxy
+        }
+
+        $start = microtime(true);
+
+        $client->connect($connection->getHost());
+        $http_request = $this->createHttpRequest($method, $url, $content, $headers);
+        $client->send($http_request);
+        $socket_response  = $client->recv();
+        $response_headers = $this->parseHttpResponse($socket_response);
+        $client->close(true);
+
+        $responseString = $response_headers['body'];
+
+        $end = microtime(true);
+
+        // TODO error
+        $errorno = 0;
+        $status  = $response_headers['status'];
+
+        if (isset($params['responseClass'])) {
+            $responseClass = $params['responseClass'];
+            $response      = new $responseClass($responseString, $status);
+        } else {
+            $response = new Response($responseString, $status);
+        }
+
+        $time = $end - $start;
+        $response->setTime($time);
+        $response->setTransportInfo([
+            'url'     => $url,
+            'headers' => $headers,
+            'body'    => $request->getBody()
+        ]);
+        //hard error
+        if ($errorno > 0) {
+            $error = curl_error($conn);
```

Comment:
This shouldn't be here

Reply:
yes. you're right. I will take care of them on the next PR.